js/tweek-local-cache/src/trie.ts

```diff
@@ -47,21 +53,49 @@ export default class Trie<TValue> {
 
   list(key?: string, index = 0): { [key: string]: TValue } {
```

Comment:
Can the shared logic of list and listEntries by merged somehow?

Reply:
Additionally, isn't list entries is just keys(list())? Is there any significant and relevant performance difference creating listEntries? or Alternatively, implement list by using listEntries with map.

Reply:
the relative thing makes it pretty difficult

Reply:
also we can't just use listEntries and map because of the way we get the values from the WeakMap.
these method actually do pretty different things

Reply:
added a walk method